### PR TITLE
Report accurate ScalaCheck seed on failing test

### PR DESCRIPTION
**Pull request by nevillelyh**
Failing seed reporting seems to be incorrect when `scalaCheckInitialSeed` is not overridden. In that case `def seed` would use `def scalaCheckInitialSeed: String = Seed.random().toBase64` and be different every time. Hence the 2 seeds seen here are actually not the ones causing test failure.

https://travis-ci.org/github/spotify/magnolify/jobs/708925430#L369

```
Failing seed: z9omj4mAsT7BcJQoh95CcofqurdkO_jIAGEWFPx7xZA=
You can reproduce this failure by adding the following override to your suite:
  override val scalaCheckInitialSeed = "5E72c0hWZ5ioOvDIj2udNT9KuVs1mGz8EBcV-3zyKgH="
```

### CODE REVIEW

**Review by olafurpg — APPROVED** [review]
Thank you for this contribution! LGTM 👍

Any thoughts @gabro ?

**Review by gabro — APPROVED** [review]
Looks great, thanks @nevillelyh for tracking this down and fixing it!

**Comment by nevillelyh** [comment]
Not sure how to test though, probably need some variation of `FrameworkTest`, extract the seed from error message & verify.

**Comment by nevillelyh** [comment]
Actually DO NOT MERGE. Let me verify if this actually works as expected first.

**Comment by olafurpg** [comment]
I am happy to merge without a test case (although it would be nice to have a test 😄 )

**Comment by nevillelyh** [comment]
Think this actually broke another case. It's hard to verify since the test is non-deterministic :sob:. At least let me verify some basic scenarios.

File: munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala
Comment (nevillelyh) [inline munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala:56]: This should be the actual "initial seed", one that's used to generate seeds for all test runs (default 100) of the same property.

File: munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala
Comment (nevillelyh) [inline munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala:61]: A single test run here. And advance seed to the next.

File: munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala
Comment (nevillelyh) [inline munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala:71]: If a test failed, say 67th out of 100, we still report the 1st seed in case the test is not pure.

File: tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala
Comment (nevillelyh) [inline tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala:27]: This test checks that when initial seed is overridden, we still generate different input per test run instead of reusing the same.

**Review by nevillelyh — COMMENTED** [review]
Added some comments. Seems to work now at least as far as my understanding of ScalaCheck & requirement goes.

File: tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala
Comment (olafurpg) [inline tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala:19]: Remove?

File: tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala
Comment (olafurpg) [inline tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala:24]: Can we move this to a separate file? I normally try to organize one test suite per file.

**Review by olafurpg — CHANGES_REQUESTED** [review]
@nevillelyh Thank you for double checking! A few minor comments, otherwise this looks ready to go

File: tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala
Comment (nevillelyh) [reply tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala]: Fixed

File: tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala
Comment (nevillelyh) [reply tests/shared/src/test/scala/munit/ScalaCheckSeedSuite.scala]: Fixed

**Review by olafurpg — APPROVED** [review]
LGTM 👍 I'll let @gabro do the merge

**Review by gabro — APPROVED** [review]
Thanks again!

**Pull request merged**
